pandas/indexes/base.py

```diff
@@ -1745,29 +1745,13 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
-        if isinstance(other, Index):
-            return self.union(other)
         return Index(np.array(self) + other)
 
     def __radd__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
         return Index(other + np.array(self))
 
     __iadd__ = __add__
 
-    def __sub__(self, other):
```

Comment:
yep, this is something that I still have to handle.

The problem is that sub/add are added for subclasses in either `_add_numericlike_set_methods_disabled` or `_add_numeric_methods` (always handled both), but in the base class we only want ``__add__`` and not `__sub__`

If I leave it in here, I think it should rather raise a TypeError? (for usage of base Index class)